.github/workflows/main.yml

```diff
@@ -0,0 +1,73 @@
+name: Master
+
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - README.md
+      - LICENSE
+
+jobs:
+  get-publish-version:
+    runs-on: ubuntu-latest
+    outputs:
+      publish-version: ${{ steps.get-publish-version.outputs.publish-version }}
+    steps:
+      - name: Get publish version
+        id: get-publish-version
+        run: |
+          LATEST_VERSION=$(curl --silent "https://api.github.com/repos/${{ github.repository }}/releases/latest" | grep '"tag_name":' | sed -E 's/.*"([^"]+)".*/\1/')
+          [  -z "$LATEST_VERSION" ] && LATEST_VERSION="0.0.0"
+          VERSION=$(echo $LATEST_VERSION | awk -F. '{$NF = $NF + 1;} 1' | sed 's/ /./g')
+          echo "::set-output name=publish-version::${VERSION}"
+
+  publish-github-release:
+    runs-on: ubuntu-latest
+    needs: [get-publish-version, build-app]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@master
+      - name: Download built app
+        uses: actions/download-artifact@v2
+        with:
+          name: built-app
+          path: build
+      - name: Create GitHub Release
+        id: create_release
+        uses: actions/create-release@latest
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ needs.get-publish-version.outputs.publish-version }}
+          release_name: Release ${{ needs.get-publish-version.outputs.publish-version }}
+          draft: false
+          prerelease: false
+      - name: Upload Release Asset
+        id: upload-release-asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.create_release.outputs.upload_url }}
+          asset_path: ./build/libs/historic-data-loader-0.0.1.jar
+          asset_name: historic-data-loader-${{ needs.get-publish-version.outputs.publish-version }}.jar
+          asset_content_type: application
+
+  build-app:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@master
+      - name: build and test
+        run: make integration-all
+      - name: java setup
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: gradle build
+        run: make local-all
+      - name: Upload app
+        uses: actions/upload-artifact@v2
+        with:
+          name: built-app
+          path: build
```

Comment:
just uploading the libs dir would be better

Reply:
I think this is a more verbose way - aids understanding.